File: thread/c/thread2.c
#include <stdio.h>
#include <unistd.h>
#include <pthread.h>


void* foo() {
    long tID = (long int) pthread_self();

    printf("PID is %d\n", getpid());
    printf("Thread ID is %ld\n", tID);

    return NULL;
}

int main(void) {
    pthread_t t1, t2, t3;
    pthread_create(&t1, NULL, foo, NULL);
    pthread_create(&t2, NULL, foo, NULL);
    pthread_create(&t3, NULL, foo, NULL);
    pthread_join(t1, NULL);
    pthread_join(t2, NULL);
    pthread_join(t3, NULL);

    return 0;
}

/*
PID is 6385
Thread ID is 6158266368
PID is 6385
Thread ID is 6158839808
PID is 6385
Thread ID is 6159413248
*/

File: thread/python/thread2.py
import os
import threading


def foo():
    print("PID:", os.getpid())
    print("Thread ID:", threading.get_native_id())


if __name__ == "__main__":
    t1 = threading.Thread(target=foo).start()
    t2 = threading.Thread(target=foo).start()
    t3 = threading.Thread(target=foo).start()


"""
PID: 6905
Thread ID: 926514
PID: 6905
Thread ID: 926515
PID: 6905
Thread ID: 926516
"""
